include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -45,73 +45,73 @@ def ForeignType : FIRRTLDialectType<CPred<"!$_self.isa<FIRRTLType>()">,
                                     "foreign type", "::mlir::Type">;
 
 def ClockType :
-  FIRRTLDialectType<CPred<"$_self.isa<ClockType>()">, "clock",
-                    "::circt::firrtl::ClockType">;
+  FIRRTLDialectType<CPred<"firrtl::type_isa<ClockType>($_self)">, "clock",
+                    "::circt::firrtl::BaseTypeAliasOr<::circt::firrtl::ClockType>">;
 
 def NonConstClockType :
-  FIRRTLDialectType<CPred<"$_self.isa<ClockType>()">, "clock",
+  FIRRTLDialectType<CPred<"firrtl::type_isa<ClockType>($_self)">, "clock",
                     "::circt::firrtl::ClockType">,
```

Comment:
Should this be `BaseTypeAliasOr` like the others, too? Please make a pass over this checking for consistency.  Looks like we may need a helper tablegen class thing to avoid some of this repetition and to make intent clear?

Reply:
BaseTypeAliasOr cannot be a buildable type so I didn't change as it wold require updating uses of NonConstClockType. But yes, I agree it's more consistent to update here. Will update.

Reply:
I mean, does this work for aliases to clock type as-is (no problems by forcing into incompatible type?).  But yes consistency is nice thank you :+1: .